### ISSUE TITLE
Addressing Feedback

### ISSUE THREAD

**Issue by jy1909** [UBC-MDS/text_processing_util_mds24#72]
Note that we have already addressed the feedback from Milestone 1 in the previous milestone.

Here, we are making changes in response to the following items from peer review which we agree that they should be addressed.

- [x] ([365df2d](https://github.com/UBC-MDS/text_processing_util_mds24/commit/365df2db011b736cbf4ee5a421dc3c7d0e10009a)) @allan8392: Python dependency is specified as python = "^3.9"; avoid setting upper limit
- [x] ([faddab4](https://github.com/UBC-MDS/text_processing_util_mds24/commit/faddab41338fc8cec5ef8265bec63da3658c0950)) @jy1909: include more badges
- [x] ([a3eeb5c](https://github.com/UBC-MDS/text_processing_util_mds24/commit/a3eeb5caa63c9f096db94e7a8a2d61188536fc16)) @nassimgha: adding "pip install" instruction in new section called "Installation"
- [x] ([0c4a5b2](https://github.com/UBC-MDS/text_processing_util_mds24/commit/0c4a5b2f679bc95389ca10643f7a2498e75a5970)) @MoNorouzi23: expand on TF-IDF and tokenizer-padding (README.md)
- [x] ([26c2295](https://github.com/UBC-MDS/text_processing_util_mds24/commit/26c2295ac24d7cdf01359cc4ffca64a8b45298c3)) @jy1909: README: in Environment section, explain how our functions are different from sklearn and Keras
- [x] ([4767835](https://github.com/UBC-MDS/text_processing_util_mds24/commit/47678354a6b73b6bea85932f82cb8d5bace4df38)) @nassimgha: Make developer installation instructions more clear. In point 4, make it clear that _it is recommended to do poetry install inside your conda environment_.
- [x] ([ee1d45f](https://github.com/UBC-MDS/text_processing_util_mds24/commit/ee1d45f34d09e74aaca86ba73321641ac8c12983)) @allan8392: Add Assert Messages for `text_clean`
- [x] ([2b17cb4](https://github.com/UBC-MDS/text_processing_util_mds24/commit/2b17cb4a1c86f367a2f8a1631a828b19c31c58ba)) @nassimgha : Add Assert Messages for `frequency_vectorizer`
- [x] ([bff23fa](https://github.com/UBC-MDS/text_processing_util_mds24/commit/bff23faca5d13b56d6c1f365255fd3eb185b8f3d)) @MoNorouzi23: Add Assert Messages for `tfidf_vectorizer`
- [x] ([cd6167c](https://github.com/UBC-MDS/text_processing_util_mds24/commit/cd6167c57ce9178e8dc4de2ee6a2c118606b0c32)) @jy1909: Add Assert Messages for `tokenizer_padding`
- [x] ([4941656](https://github.com/UBC-MDS/text_processing_util_mds24/commit/49416560150eec61a2200dfd7f89ce83e14ebaab)) @jy1909: Make it clear that one needs to be in the right environment to run `pytest`

I recommend creating a separate branch for each bullet point. I will create separate issues for each bullet point later.

Reminder: to make it clear to the TAs that we are making changes to address the feedback, write something like this in the commit messages: `fix: Feedback addressed by ...`

**Update** (Jan 31): Milestone 2 was just released yesterday. We should add the assert messages for our tests. Hence, I added one more task for each function here.

**Update 2** (Jan 31): I checked our CI-CD on the main branch, and our 33 tests all passed. There are comments about one of our tests not passing from the reviewers. I think what likely have happened is that they are not in the conda environment in which they ran `poetry install`. This is exactly what happened to me earlier today (i.e. I got one test error) when I forgot to switch environments. I will make this more clear in the Testing section of `README.md`.

**Comment by jy1909** [comment]
I created all the issues for addressing the feedback from the peer review.

**Comment by MoNorouzi23** [comment]
I added more information on the README file for 2 functions!

**Comment by jy1909** [comment]
Changes to the environment section in README to explain how our functions are different were merged.

**Comment by jy1909** [comment]
I just made 2 updates in my original comment (Jan 31). Please check them out.

**Comment by jy1909** [comment]
All the items in our list relating to feedback have been addressed.

**Comment by nassimgha** [comment]
Everything has been addressed, thanks.
